File: create_favicon.js
const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

const inputPath = path.join(process.cwd(), 'public', 'logo_clean.png');
const outputPath = path.join(process.cwd(), 'src', 'app', 'icon.png');

async function cropFavicon() {
    try {
        const image = sharp(inputPath);
        const metadata = await image.metadata();

        // Assume icon is on the left and square-ish.
        // We'll crop a square from the left side.
        const height = metadata.height;
        const width = metadata.width;

        // If it's a wide logo, crop left square.
        // If it's effectively square already, just resize.
        let extractRegion;

        if (width > height * 1.2) {
            // Wide logo: Crop left square
            extractRegion = { left: 0, top: 0, width: height, height: height };
        } else {
            // Square or portrait: Use as is (maybe resize if needed, but Next.js handles that)
            extractRegion = null;
        }

        let pipeline = image;
        if (extractRegion) {
            pipeline = pipeline.extract(extractRegion);
        }

        // Resize to 512x512 for good measure (standard icon size)
        pipeline = pipeline.resize(512, 512).png();

        await pipeline.toFile(outputPath);
        console.log("Favicon created at " + outputPath);

    } catch (error) {
        console.error("Error creating favicon:", error);
        process.exit(1);
    }
}

cropFavicon();

File: create_favicon.py
from PIL import Image
import os

def create_favicon(input_path, output_path):
    try:
        img = Image.open(input_path)
        # Assuming the logo is wide (TEXT + ICON usually), or ICON + TEXT.
        # User said "Front part of the logo". 
        # Usually logos are square if just icon, but if "Beaueli" is text, maybe the icon is on the left?
        # Let's inspect the image size first implicitly by cropping a square from the left.
        
        width, height = img.size
        # Crop the left-most square area (assuming icon is there)
        # If the image is 500x100, we crop 100x100 from 0,0.
        
        crop_size = min(width, height)
        # Actually if it's a wide logo, the icon is likely on the left.
        # Let's crop a square from the left.
        
        left = 0
        top = 0
        right = height # Make it a square based on height
        bottom = height
        
        if width > height:
             # Wide logo
             cropped = img.crop((left, top, right, bottom))
        else:
             # Vertical or square logo, just resize
             cropped = img
        
        # Resize to standard favicon size (e.g. 512x512 for high res icon.png)
        cropped = cropped.resize((192, 192), Image.Resampling.LANCZOS)
        
        cropped.save(output_path)
        print(f"Successfully created favicon at {output_path}")

    except Exception as e:
        print(f"Error: {e}")

# Run
create_favicon('/Users/samuelkim/Desktop/Renewal Beaueli/public/logo_clean.png', '/Users/samuelkim/Desktop/Renewal Beaueli/src/app/icon.png')
